apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -39,6 +39,10 @@ const (
 	// an immediate requeue)
 	PausedAnnotation = "baremetalhost.metal3.io/paused"
 
+	// Deregistered is the annotation which stops provisioner management of the host
+	// unlike in the paused case, the host status may be updated
+	DeregisteredAnnotation = "baremetalhost.metal3.io/deregistered"
```

Comment:
Argh! If only `Unmanaged` and `paused` weren't already in use! :laughing: 

After thinking about the name I reached the conclusion that `deregistered` reflects the internal state quite well (and it was definitely confusing having a State and OperationalStatus both called `unmanaged` after I moved the code into the same file).

Then it seemed weird to dislocate that naming convention at the annotation level, so I just renamed it and personally I think deregistered is an OK name given that the two most obvious alternatives are already in use?

With the benefit of hindsight we should perhaps have called the `paused` annotation `frozen` or something, but ultimately we just have to pick a name then document what it does.

I don't want to burn a bunch more cycles on the naming, but does anyone else have strong opinions on this?

Reply:
Not a strong one, my inclination would be more on `detach` rather then `deregister` - just because we already have a `Registering` state, and to me sounds as something related. `detach` gives me more the idea of a BMH still alive but "detached" from an Ironic node.

Reply:
BTW I intended "disengage" in the sense of a gearbox. We're basically putting it in neutral so you can attach another engine for a bit. It's quite a nice analogy, at least if you get a chance to explain it to the user ;D
My only concern with "detach", and why I listed it in 3rd place, is that it is more prone to conflicts with other things like networks or volumes (i.e. it risks repeating the same mistake we have made with "paused" and "unmanaged"). I still prefer it to "deregister" though.
"Deregister" makes sense if you understand the inside baseball (i.e. that there's a separate volatile data store, that existence of an entry in this datastore means that the power state will be continuously monitored, and that when this data store is cleared we silently re-register existing hosts in it.) However, to a user looking in from the outside and seeing only the API, I think it makes little sense at best, and at worst is actively misleading by implying that it resets the state back to the time before "registration".

Reply:
Ugh.  Naming is hard :disappointed: 

@dhellmann @maelk @fmuyassarov - any preference before I rename this again?

FWIW my vote is for `detached` - I take the point about potential overlap with networking/storage things @zaneb but in the context of the BMH resource that's probably not very likely, right?

Reply:
Changed to `detached` in the latest revision

Reply:
> that's probably not very likely, right?

:rofl:

metal3-io/metal3-docs#132

Reply:
> > that's probably not very likely, right?
> 
> rofl
> 
> [metal3-io/metal3-docs#132](https://github.com/metal3-io/metal3-docs/pull/132)

Thanks I missed that - and it adds a ports API to the BMH spec - I'm not sure that's a good idea, I was thinking any such additions would have new CRDs, and thus not overlap with BMH things, but it seems I've been wildly optimistic there.